diff -up dmenu-4.6/dmenu.c dmenu-4.6-normprompt/dmenu.c
--- dmenu-4.6/dmenu.c	2015-11-09 07:42:21.000000000 +0900
+++ dmenu-4.6-normprompt/dmenu.c	2016-07-18 13:21:39.159763099 +0900
@@ -125,7 +125,7 @@ drawmenu(void)
 	drw_rect(drw, 0, 0, mw, mh, 1, 1, 1);
 
 	if (prompt && *prompt) {
-		drw_setscheme(drw, &scheme[SchemeSel]);
+		drw_setscheme(drw, &scheme[SchemeNorm]);
 		drw_text(drw, x, 0, promptw, bh, prompt, 0);
 		x += promptw;
 	}
